File: dotnet-elasticemail-examples/Examples/ScheduledSend.cs
namespace ElasticEmailExamples;

using ElasticEmail.Api;
using ElasticEmail.Client;
using ElasticEmail.Model;

public static class ScheduledSend
{
    public static async Task RunAsync()
    {
        var emailsApi = new EmailsApi(Ee.Config());

        // TimeOffset delays delivery by N minutes from now. Maximum is 35 days (50400 minutes).
        // Elastic Email has no cancel endpoint for a single delayed email, so pick the offset carefully.
        const int delayMinutes = 60;
        var scheduledFor = DateTime.UtcNow.AddMinutes(delayMinutes);

        try
        {
            var result = await emailsApi.EmailsTransactionalPostAsync(new EmailTransactionalMessageData(
                recipients: new TransactionalRecipient(to: new List<string> { Ee.To }),
                content: new EmailContent(
                    from: Ee.From,
                    subject: "Scheduled Email",
                    body: new List<BodyPart>
                    {
                        new BodyPart(BodyContentType.HTML,
                            $"<h1>Scheduled Email</h1><p>This email was scheduled for {scheduledFor:O}.</p>"),
                    }),
                options: new Options(timeOffset: delayMinutes)));

            Console.WriteLine($"Email scheduled for {scheduledFor:O}");
            Console.WriteLine($"Transaction ID: {result.TransactionID}");
            Console.WriteLine($"Message ID: {result.MessageID}");
        }
        catch (ApiException e)
        {
            Ee.Fail("schedule email", e);
        }
    }
}

File: rust-elasticemail-examples/examples/scheduled_send.rs
use rust_elasticemail_examples::{config, from, print_api_error, to};
use ElasticEmail::apis::emails_api;
use ElasticEmail::models::{
    BodyContentType, BodyPart, EmailContent, EmailTransactionalMessageData, Options, TransactionalRecipient,
};

#[tokio::main]
async fn main() {
    let config = config();

    // TimeOffset delays delivery by N minutes from now. Maximum is 35 days (50400 minutes).
    // Elastic Email has no cancel endpoint for a single delayed email, so pick the offset carefully.
    let delay_minutes: i32 = 60;
    let scheduled_for = (chrono::Utc::now() + chrono::Duration::minutes(delay_minutes as i64)).to_rfc3339();

    let content = EmailContent {
        subject: Some("Scheduled Email".to_string()),
        body: Some(vec![BodyPart {
            content: Some(format!("<h1>Scheduled Email</h1><p>This email was scheduled for {}.</p>", scheduled_for)),
            ..BodyPart::new(BodyContentType::Html)
        }]),
        ..EmailContent::new(from())
    };

    let data = EmailTransactionalMessageData {
        options: Some(Box::new(Options { time_offset: Some(Some(delay_minutes)), ..Options::new() })),
        ..EmailTransactionalMessageData::new(TransactionalRecipient::new(vec![to()]), content)
    };

    match emails_api::emails_transactional_post(&config, data).await {
        Ok(result) => {
            println!("Email scheduled for {}", scheduled_for);
            println!("Transaction ID: {}", result.transaction_id.unwrap_or_default());
            println!("Message ID: {}", result.message_id.unwrap_or_default());
        }
        Err(e) => {
            print_api_error("scheduling email", &e);
            std::process::exit(1);
        }
    }
}
